erc20-bridge-token/test/BridgeToken.js

```diff
@@ -0,0 +1,37 @@
+const { expect } = require('chai');
+const truffleAssert = require('truffle-assertions');
+const BridgeToken = artifacts.require('BridgeToken.sol');
+
+const { toWei, fromWei, hexToBytes } = web3.utils;
+
+contract('BridgeToken', function ([_, addr1]) {
+    beforeEach(async function () {
+        
+    });
+
+    it('can create empty token', async function() {
+        const token = await BridgeToken.new("", "", 0);
+        expect(await token.name()).to.equal("");
+        expect(await token.symbol()).to.equal("");
+        expect((await token.decimals()).toString()).to.equal("0");
+        expect((await token.metadataLastUpdated()).toString()).to.equal("0");
+    });
+
+    it('can update metadata', async function() {
+        const token = await BridgeToken.new("", "", 0);        
+        let block = await web3.eth.getBlock("latest");
+        await token.set_metadata("NEAR ERC20", "NEAR", "18", String(block.number));
+
+        expect(await token.name()).to.equal("NEAR ERC20");
+        expect(await token.symbol()).to.equal("NEAR");
+        expect((await token.decimals()).toString()).to.equal("18");
+        expect((await token.metadataLastUpdated()).toString()).to.equal(String(block.number));
+    });
```

Comment:
Please also add tests for controller-related checks (preferably with ACL) and factory-related tests for every method.

Reply:
[Latest test version ](https://github.com/aurora-is-near/rainbow-token-connector/pull/99/commits/0326f9effa784ca78073c7460c7cb903526ee4fb)